File: app/src/main/java/com/worldventures/dreamtrips/modules/reptools/service/command/LikeSuccessStoryCommand.java
package com.worldventures.dreamtrips.modules.reptools.service.command;

import com.worldventures.dreamtrips.R;
import com.worldventures.dreamtrips.api.success_stories.LikeSuccessStoryHttpAction;
import com.worldventures.dreamtrips.core.api.action.ApiActionCommand;

import io.techery.janet.command.annotations.CommandAction;

@CommandAction
public class LikeSuccessStoryCommand extends ApiActionCommand<LikeSuccessStoryHttpAction, Void> {

   int id;

   public LikeSuccessStoryCommand(int id) {
      this.id = id;
   }

   @Override
   protected Void mapHttpActionResult(LikeSuccessStoryHttpAction httpAction) {
      return null;
   }

   @Override
   protected LikeSuccessStoryHttpAction getHttpAction() {
      return new LikeSuccessStoryHttpAction(id);
   }

   @Override
   protected Class<LikeSuccessStoryHttpAction> getHttpActionClass() {
      return LikeSuccessStoryHttpAction.class;
   }

   @Override
   public int getFallbackErrorMessage() {
      return R.string.error_fail_to_like_success_story;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/reptools/command/LikeSuccessStoryCommand.kt
package com.worldventures.dreamtrips.social.service.reptools.command

import com.worldventures.core.service.command.api_action.ApiActionCommand
import com.worldventures.dreamtrips.R
import com.worldventures.dreamtrips.api.success_stories.LikeSuccessStoryHttpAction

import io.techery.janet.command.annotations.CommandAction

@CommandAction
class LikeSuccessStoryCommand(private val id: Int) : ApiActionCommand<LikeSuccessStoryHttpAction, Int>() {

   override fun getHttpAction(): LikeSuccessStoryHttpAction = LikeSuccessStoryHttpAction(id)

   override fun getHttpActionClass(): Class<LikeSuccessStoryHttpAction> = LikeSuccessStoryHttpAction::class.java

   override fun getFallbackErrorMessage() = R.string.error_fail_to_like_success_story

   override fun mapCommandResult(httpCommandResult: Int?) = id
}
